force-app/main/default/classes/OpportunityTriggerHandler.cls

```diff
@@ -0,0 +1,171 @@
+public with sharing class OpportunityTriggerHandler extends TriggerHandler{
+ 
+    private List<Opportunity> newOppList;
+    private List<Opportunity> oldOppList;
+    private Map<Id, Opportunity> newOppMap;
+    private Map<Id, Opportunity> oldOppMap;
+    public static Boolean hasRun = false;
+    
+
+    public OpportunityTriggerHandler(){
+        this.newOppList = (List<Opportunity>)Trigger.new;
+        this.oldOppList = (List<Opportunity>)Trigger.old;
+        this.newOppMap = (Map<Id,Opportunity>)Trigger.newMap;
+        this.oldOppMap = (Map<Id,Opportunity>)Trigger.oldMap;
+    }
+
+
+    public override void beforeInsert(){
+        // Set default Type for new Opportunities
+        for(Opportunity opp : this.newOppList){
```

Comment:
You can also consider moving the logic inside your handler methods to an OpportunityHelper class